src/game/sound_samples.c

```diff
@@ -10,14 +10,92 @@
 #include "system/nth_alloc.h"
 
 // TODO(#863): Sound_samples is not implemented
+// TODO: Volume control?
 
 struct Sound_samples
 {
     Lt *lt;
+    SDL_AudioDeviceID dev;
+    uint8_t **audio_buf_array;
+    uint32_t *audio_buf_size_array;
     size_t samples_count;
     int paused;
 };
 
+static 
+int init_buffer_and_device(Sound_samples *sound_samples, 
+                           const char *sample_files[]) 
+{
+    // TODO: Select audio specification from a menu
+    // TODO: Use a seperate callback function
+    SDL_AudioSpec destination_spec = { // stereo float32 44100Hz
+        .format = AUDIO_F32,
+        .channels = 2,
+        .freq = 44100
+    }, actual;
+    // TODO: a return value by SDL_GetNumAudioDevices that is <= 0 may not indicate an error
+    if (SDL_GetNumAudioDevices(0) <= 0) {
+        log_fail("No audio in 2019 LULW\n");
+        return -1;
+    }
+    
+    sound_samples->audio_buf_array = PUSH_LT(sound_samples->lt, nth_calloc(sound_samples->samples_count, sizeof(uint8_t*)), free);
+    if (sound_samples->audio_buf_array == NULL) {
+        log_fail("Failed to allocate memory for audio buffer pointer array\n");
+        return -1;
+    }
+    sound_samples->audio_buf_size_array = PUSH_LT(sound_samples->lt, nth_calloc(sound_samples->samples_count, sizeof(uint32_t)), free);
+    if (sound_samples->audio_buf_size_array == NULL) {
+        log_fail("Failed to allocate memory for audio buffer size array\n");
+        return -1;
+    }
+    for (size_t i = 0; i < sound_samples->samples_count; ++i) {
+        uint8_t *wav_buf; uint32_t wav_buf_len; SDL_AudioSpec wav_spec;
+
+        log_info("Loading audio file %s...\n", sample_files[i]);
+        if (SDL_LoadWAV(sample_files[i], &wav_spec, &wav_buf, &wav_buf_len) == NULL) {
+            log_fail("Load WAV file failed: %s\n", SDL_GetError());
+            return -1;
+        }
+        PUSH_LT(sound_samples->lt, wav_buf, SDL_FreeWAV);
+        SDL_AudioCVT cvt;
+        int result = SDL_BuildAudioCVT(&cvt, wav_spec.format, (uint8_t)wav_spec.channels, (int)wav_spec.freq, 
+                          destination_spec.format, (uint8_t)destination_spec.channels, (int)destination_spec.freq);
+        if (result < 0) {
+            log_fail("SDL_BuildAudioCVT failed: %s\n", SDL_GetError());
+            return -1;
+        } else if (result == 0) { // no need to do conversion
+            sound_samples->audio_buf_array[i] = wav_buf;
+            sound_samples->audio_buf_size_array[i] = wav_buf_len;
+        } else {
+            cvt.len = (int)wav_buf_len;
+            cvt.buf = PUSH_LT(sound_samples->lt, malloc((size_t)(cvt.len * cvt.len_mult)), free);
+            if (cvt.buf == NULL) {
+                log_fail("Allocating buffer for conversion failed\n");
+                return -1;
+            }
+            memcpy(cvt.buf, wav_buf, (size_t)cvt.len);
+            SDL_FreeWAV(wav_buf);
+            RELEASE_LT(sound_samples->lt, wav_buf);
```

Comment:
You can actually collapse this to operations into `SDL_FreeWAV(RELEASE_LT(sound_samples->lt, wav_buf))`.